clang/test/CIR/CodeGen/no-prototype.c

```diff
@@ -35,7 +35,9 @@ int test1(int x) {
 int noProto2();
 int test2(int x) {
   return noProto2(x);
-  // CHECK: %{{.+}} = cir.call @noProto2(%{{[0-9]+}}) : (!s32i) -> !s32i
+  // CHECK:  [[GGO:%.*]] = cir.get_global @noProto2 : cir.ptr <!cir.func<!s32i (!s32i)>>
+  // CHECK:  [[CAST:%.*]] = cir.cast(bitcast, %3 : !cir.ptr<!cir.func<!s32i (!s32i)>>), !cir.ptr<!cir.func<!s32i (!s32i)>>
```

Comment:
We probably shouldn't be emitting this specific bitcast because it's redundant. The other version of `createBitcast` is already ignoring unnecessary bitcasts, if you implement the version with 2 args to call in the version with 3 args instead (or unify them somehow), you could fix this without having to add an `if` during CIRGen.

Reply:
We can't avoid emitting this bitcasts in such simple way. The issue is that we build call of functions without prototype. So the callee type may be unknown is this place.
I updated the patch to remove redundant bitcasts in `ReplaceUsesOfNonProtoTypeWithRealFunction`. Pls, check it

Reply:
> The issue is that we build call of functions without prototype. So the callee type may be unknown is this place.

Oh, I see. At that point in the code we don't have all type info just yet. Thanks for clarifying.

> I updated the patch to remove redundant bitcasts in ReplaceUsesOfNonProtoTypeWithRealFunction. Pls, check it

I rather we stay simple and fast during CIRGen, later passes should understand that such bitcasts are trivial and get rid of them. So please get back to the previous version of the PR.

Reply:
PR was reverted to the previous version.